Marlin/src/lcd/menu/menu_info.cpp

```diff
@@ -185,41 +198,87 @@ void menu_info_board() {
 //
 // About Printer > Printer Info
 //
-void menu_info_printer() {
-  if (ui.use_click()) return ui.goto_previous_screen();
-  START_SCREEN();
-  STATIC_ITEM(MSG_MARLIN, true, true);                             // Marlin
-  STATIC_ITEM(SHORT_BUILD_VERSION, true);                          // x.x.x-Branch
-  STATIC_ITEM(STRING_DISTRIBUTION_DATE, true);                     // YYYY-MM-DD HH:MM
-  STATIC_ITEM(MACHINE_NAME, true);                                 // My3DPrinter
-  STATIC_ITEM(WEBSITE_URL, true);                                  // www.my3dprinter.com
-  STATIC_ITEM(MSG_INFO_EXTRUDERS ": " STRINGIFY(EXTRUDERS), true); // Extruders: 2
-  #if ENABLED(AUTO_BED_LEVELING_3POINT)
-    STATIC_ITEM(MSG_3POINT_LEVELING, true);                        // 3-Point Leveling
-  #elif ENABLED(AUTO_BED_LEVELING_LINEAR)
-    STATIC_ITEM(MSG_LINEAR_LEVELING, true);                        // Linear Leveling
-  #elif ENABLED(AUTO_BED_LEVELING_BILINEAR)
-    STATIC_ITEM(MSG_BILINEAR_LEVELING, true);                      // Bi-linear Leveling
-  #elif ENABLED(AUTO_BED_LEVELING_UBL)
-    STATIC_ITEM(MSG_UBL_LEVELING, true);                           // Unified Bed Leveling
-  #elif ENABLED(MESH_BED_LEVELING)
-    STATIC_ITEM(MSG_MESH_LEVELING, true);                          // Mesh Leveling
+#if DISABLED(LCD_INFO_PRINTER_SHOWS_BOOTSCREEN)
+  void menu_info_printer() {
+    if (ui.use_click()) return ui.goto_previous_screen();
+    START_SCREEN();
+    STATIC_ITEM(MSG_MARLIN, true, true);                             // Marlin
+    STATIC_ITEM(SHORT_BUILD_VERSION, true);                          // x.x.x-Branch
+    STATIC_ITEM(STRING_DISTRIBUTION_DATE, true);                     // YYYY-MM-DD HH:MM
+    STATIC_ITEM(MACHINE_NAME, true);                                 // My3DPrinter
+    STATIC_ITEM(WEBSITE_URL, true);                                  // www.my3dprinter.com
+    STATIC_ITEM(MSG_INFO_EXTRUDERS ": " STRINGIFY(EXTRUDERS), true); // Extruders: 2
+    #if ENABLED(AUTO_BED_LEVELING_3POINT)
+      STATIC_ITEM(MSG_3POINT_LEVELING, true);                        // 3-Point Leveling
+    #elif ENABLED(AUTO_BED_LEVELING_LINEAR)
+      STATIC_ITEM(MSG_LINEAR_LEVELING, true);                        // Linear Leveling
+    #elif ENABLED(AUTO_BED_LEVELING_BILINEAR)
+      STATIC_ITEM(MSG_BILINEAR_LEVELING, true);                      // Bi-linear Leveling
+    #elif ENABLED(AUTO_BED_LEVELING_UBL)
+      STATIC_ITEM(MSG_UBL_LEVELING, true);                           // Unified Bed Leveling
+    #elif ENABLED(MESH_BED_LEVELING)
+      STATIC_ITEM(MSG_MESH_LEVELING, true);                          // Mesh Leveling
+    #endif
+    END_SCREEN();
+  }
+#else
+  void menu_show_marlin_bootscreen() {
+    if (ui.use_click()) { ui.goto_previous_screen_no_defer(); }
+    ui.draw_marlin_bootscreen();
```

Comment:
While you can sort of get away with this, it isn't a supported way to display the boot screen. Screen handlers are called in a loop by the LCD handler, with one stripe per call on graphical displays. Meanwhile, these boot screen functions also do a full display loop to draw the whole boot screen. I believe these end up only drawing once at first, but rotating the knob may produce some strange results since this causes the screen handler to be called on each tick. Probably should do some experimentation and see if it can be made to break or "glitch out."

Reply:
Oh, okay. I'll take a look and see if this can be improved.

Reply:
@thinkyhead: This works without LCD glitches, but if the user selects the menu option during a print, it could potentially cause a goober in the print. I'll refactor the code so that when it is called from the menu code, the outer u8g loop and delays aren't used.

Reply:
Okay, I pushed a fix.